src/aliceVision/keyframe/KeyframeSelector.cpp

```diff
@@ -25,452 +25,711 @@ namespace keyframe {
 
 /**
  * @brief Get a random int in order to generate uid.
- * @warning The random don't use a repeatable seed to avoid conflicts between different launches on different data sets.
+ * @warning The random doesn't use a repeatable seed to avoid conflicts between different launches on different data sets
  * @return int between 0 and std::numeric_limits<int>::max()
  */
 int getRandomInt()
 {
-  std::random_device rd;  // will be used to obtain a seed for the random number engine
-  std::mt19937 randomTwEngine(rd()); // standard mersenne_twister_engine seeded with rd()
-  std::uniform_int_distribution<> randomDist(0, std::numeric_limits<int>::max());
-  return randomDist(randomTwEngine);
+    std::random_device rd;  // will be used to obtain a seed for the random number engine
+    std::mt19937 randomTwEngine(rd());  // standard mersenne_twister_engine seeded with rd()
+    std::uniform_int_distribution<> randomDist(0, std::numeric_limits<int>::max());
+    return randomDist(randomTwEngine);
+}
+
+/**
+ * @brief Read an image from a feed provider into a grayscale OpenCV matrix, and rescale it if a size is provided.
+ * @param[in] feed The feed provider
+ * @param[in] width The width to resize the input image to. The height will be adjusted with respect to the size ratio.
+ *                  There will be no resizing if this parameter is set to 0
+ * @return An OpenCV Mat object containing the image
+ */
+cv::Mat readImage(dataio::FeedProvider &feed, std::size_t width = 0)
```

Comment:
maybe place the declaration and documentation in header file ?

Reply:
I added it to the KeyframeSelector class as a private member. Done.